### ISSUE TITLE
Error saving merged models: "KeyError: "sd_merge_recipe"

### ISSUE THREAD

**Issue by CCpt5** [wkpark/sd-webui-model-mixer#8]
The save current merge model feature hasn't yet worked for me. In earlier versions I'd end up w/ a checkpoint that would come up as mostly "junk" data when loaded into the "[model toolkit](https://github.com/arenasys/stable-diffusion-webui-model-toolkit)" extension. I checked them there as they were odd sizes and did not work in the app or for training.  Those files would be generated if I unticked "Safetensors". Leaving that ticked gave a different error and failed to generate a checkpoint (sorry I don't have a log of that).  

At the moment though when I try to save a model that has been created in an open session I'm getting the error "KeyError: 'sd_merge_recipe'":

```
 `To create a public link, set `share=True` in `launch()`.
Startup time: 1.7s (load scripts: 0.9s, create ui: 0.3s, gradio launch: 0.3s).
Traceback (most recent call last):
  File "D:\Stable-Diffusion-Webui-Dev\sdxl\stable-diffusion-webui\venv\lib\site-packages\gradio\routes.py", line 488, in run_predict
    output = await app.get_blocks().process_api(
  File "D:\Stable-Diffusion-Webui-Dev\sdxl\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 1431, in process_api
    result = await self.call_function(
  File "D:\Stable-Diffusion-Webui-Dev\sdxl\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 1103, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "D:\Stable-Diffusion-Webui-Dev\sdxl\stable-diffusion-webui\venv\lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "D:\Stable-Diffusion-Webui-Dev\sdxl\stable-diffusion-webui\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
  File "D:\Stable-Diffusion-Webui-Dev\sdxl\stable-diffusion-webui\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "D:\Stable-Diffusion-Webui-Dev\sdxl\stable-diffusion-webui\venv\lib\site-packages\gradio\utils.py", line 707, in wrapper
    response = f(*args, **kwargs)
  File "D:\Stable-Diffusion-Webui-Dev\sdxl\stable-diffusion-webui\extensions\sd-webui-model-mixer\scripts\model_mixer.py", line 811, in save_current_model
    metadata["sd_merge_recipe"] = json.dumps(metadata["sd_merge_recipe"])
KeyError: 'sd_merge_recipe'`
```

**Comment by wkpark** [comment]
thanks for your reporting,

A merged model by Model mixer always sets the metadata, I guess you have tried to save is not a valid merge,
(Internally, a loaded model is stored in the `shared.sd_model`, and Model Mixer wrongly checked its validity (bug))
in this case, the correct behavior is to print out a "Current checkpoint is not a merged one." warning message.

fixed by commit 5fc7a42731d

**Comment by wkpark** [comment]
same issue here, this is another issue.
my case: use `--medvram-sdxl`, in this case, internal `shared.sd_model` is the loaded model. but it's sizes are mismatched.

**Comment by wkpark** [comment]
fixed! #11 will be merged shortly.

**Comment by wkpark** [comment]
https://github.com/wkpark/sd-webui-model-mixer/pull/18
for some cases, merged models can't be saved correctly. in these cases, you can use "save model" option in the "Settings"->"Model Mixer".
